app/scripts/components/exploration/components/datasets/colorRangeSlider.tsx

```diff
@@ -0,0 +1,270 @@
+import React, { useCallback, useEffect, useState, useRef } from 'react';
+
+import './color-range-slider.scss';
+import { colorMapScale } from '$components/exploration/types.d.ts';
+
+import { USWDSTextInput } from '$components/common/uswds';
+
+interface ColorrangeRangeSlideProps {
+  // Absolute minimum of color range
+  min: number;
+
+  // Absolute maximum of color range
+  max: number;
+
+  // Previously selected minimum and maximum of colorRangeScale
+  colorMapScale: colorMapScale | undefined;
+
+  // Update colorRangeScale
+  setColorMapScale: (colorMapScale: colorMapScale) => void;
+}
+
+export function ColorRangeSlider({
+  min,
+  max,
+  colorMapScale,
+  setColorMapScale
+}: ColorrangeRangeSlideProps) {
+  const [minVal, setMinVal] = useState(
+    colorMapScale?.min ? colorMapScale.min : min
+  );
+  const [maxVal, setMaxVal] = useState(
+    colorMapScale?.min ? colorMapScale.max : max
+  );
+  const [inputError, setInputError] = useState({
+    min: false,
+    max: false,
+    largerThanMax: false,
+    lessThanMin: false
+  });
+
+  const minValRef = useRef<string | number>(
+    colorMapScale?.min ? colorMapScale.min : min
+  );
+  const maxValRef = useRef<string | number>(
+    colorMapScale?.max ? colorMapScale.max : max
+  );
+  const range = useRef<HTMLDivElement>(null);
+  const middleMarker = useRef<HTMLDivElement>(null);
+
+  // Convert to percentage
+  const getPercent = useCallback(
+    (value) => Math.round(((value - min) / (max - min)) * 100),
+    [min, max]
+  );
+  //Calculate the range
```

Comment:
Comments like this are a bit obsolete, especially when the utility function you wrote looks pretty clean and self-explanatory ;) Same for the `getPercent` above.